src/python/pants/backend/docker/goals/package_image.py

```diff
@@ -409,7 +408,7 @@ async def build_docker_image(
 
     image_tags_requests = union_membership.get(DockerImageTagsRequest)
     additional_image_tags = await concurrently(
-        Get(DockerImageTags, DockerImageTagsRequest, image_tags_request_cls(wrapped_target.target))
+        image_tags_request_cls.image_tags_rule()(image_tags_request_cls(wrapped_target.target))
```

Comment:
We'd have to do some tapdancing here if we wanted to continue to support old-style Gets during a deprecation period, as well as this new by-name thing. Since unions are a big part of our plugin mechanism (e.g., for plugging in a custom linter or whatever) that might be a good thing to do.